datasets/doc/source/how-to-use-with-local-data.rst

```diff
@@ -0,0 +1,226 @@
+Use with Local Data
+===================
+
+You can partition data from local files and in-memory data in
+``Flower Datasets`` library using ``Partitioner`` s.
+This guide details how to load and partition your dataset created from:
+
+* local files: CSV, JSON, image, audio data,
+* in-memory data : dictionary, list, pd.DataFrame, np.ndarray.
+
+
+General Overview
+----------------
+FederatedDataset is an abstraction that performs all the steps to prepare the dataset for FL experiments: downloading, preprocessing (including resplitting), and partitioning.
+However, the partitioning happens thanks to the ``partitioners`` of type ``str: Partitioner`` (meaning a dictionary mapping the split name of the dataset to the ``Partitioner``).
+
+You can also use ``Partitioner`` s alone without relying on the FederatedDataset (skipping the data download part). A ``Partitioner`` is also not concerned if the data is downloaded from the Hugging Face Hub or created from another source (e.g., loaded locally). The only crucial point is that the dataset you assign to the ``Partitioner`` has to be of the Dataset type (datasets.Dataset).
```

Comment:
```suggestion
You can also use ``Partitioner`` s alone without relying on the ``FederatedDataset`` (skipping the data download part). A ``Partitioner`` is also not concerned if the data is downloaded from the Hugging Face Hub or created from another source (e.g., loaded locally). The only crucial point is that the dataset you assign to the ``Partitioner`` has to be of type `datasets.Dataset <https://huggingface.co/docs/datasets/en/package_reference/main_classes#datasets.Dataset>`_.
```

Reply:
One small thing, i'm not 100% happy with "``Partitioner`` s", looks strange. Maybe we can rewrite it to avoid using plural?

Reply:
Ok, I'll go with "You can also use any Partitioner"